File: searches/binary_search/c/binary_search.c
int	binary_search(int array[], int item, int len)
{
	int	low;
	int	high;
	int	mid;
	int	current_element;

	low = 0;
	high = len - 1;
	while (low <= high)
	{
		mid = (low + high) / 2;
		current_element = array[mid];
		if (item == current_element)
			return (mid);
		else if (item < current_element)
			high = mid - 1;
		else
			low = mid + 1;
	}
	return (-1);
}

File: searches/binary_search/swift/binary_search.swift
func binarySearch(sortedCollection: [Int], item: Int) -> Int? {
    var low: Int = 0
    var high: Int = sortedCollection.count - 1
    var mid: Int
    var currentElement: Int

    while low <= high {
        mid = (low + high) / 2
        currentElement = sortedCollection[mid]

        if currentElement == item {
            return mid
        } else if currentElement > item {
            high = mid - 1
        } else {
            low = mid + 1
        }
    }
    return nil
}
